test/functional/repository/find-options/repository-find-options.ts

```diff
@@ -51,4 +52,75 @@ describe("repository > find options", () => {
 
     })));
 
+    it("should select specific columns", () => Promise.all(connections.map(async connection => {
+
+        const category = new Category();
+        category.name = "Bears";
+        await connection.manager.save(category);
+
+        const categories = [category];
+        const promises = [];
+        const photos = [];
+        for (let i = 1; i < 10; i++) {
+            const photo = new Photo();
+            photo.id = i;
+            photo.name = `Me and Bears ${i}`;
+            photo.description = `I am near bears ${i}`;
+            photo.filename = `photo-with-bears-${i}.jpg`;
+            photo.views = 10;
+            photo.isPublished = false;
+            photo.categories = categories;
+            photos.push(photo);
+            promises.push(connection.manager.save(photo));
+        }
+
+        await Promise.all(promises);
+
+        const loadedPhoto = await connection.getRepository(Photo).findOne({
+            select: ["name"],
+            where: {
+                id: 5
+            },
+        });
+
+        const loadedPhotos1 = await connection.getRepository(Photo).find({
+            select: ["filename", "views"],
+        });
+
+        const loadedPhotos2 = await connection.getRepository(Photo).find({
+            select: ["id", "name", "description"],
+            relations: ["categories"],
+        });
+
+        // const loadedPhotos3 = await connection.getRepository(Photo).createQueryBuilder("photo")
+        //     .select(["photo.name", "photo.description"])
+        //     .addSelect(["category.name"])
+        //     .leftJoin("photo.categories", "category")
+        //     .getMany();
+
+        expect(loadedPhoto).to.be.eql({
+            name: "Me and Bears 5"
+        });
+
+        expect(loadedPhotos1).to.have.deep.members(photos.map(photo => ({
+            filename: photo.filename,
+            views: photo.views,
+        })));
+
+        expect(loadedPhotos2).to.have.deep.members(photos.map(photo => ({
+            id: photo.id,
+            name: photo.name,
+            description: photo.description,
+            categories,
+        })));
+
+        // expect(loadedPhotos3).to.have.deep.members(photos.map(photo => ({
```

Comment:
I commented out this test because I'm not sure how to handle this case. If we allow virtual primary columns for this case, then the test passes. But it might be unexpected behaviour for users of querybuilder to have ids inserted into their queries when not specified. I think we should only use the virtual columns when coming from the repository